mlflow/langchain/api_request_parallel_processor.py

```diff
@@ -336,6 +366,56 @@ def _try_transform_response_to_chat_format(response):
         else:
             return transformed_response.model_dump(mode="json")
 
+    @staticmethod
+    def _try_transform_response_iter_to_chat_format(chunk_iter):
+        from langchain_core.messages.ai import AIMessageChunk
+
+        is_pydantic_v1 = Version(pydantic.__version__) < Version("2.0")
+
+        def _gen_converted_chunk(message_content, finish_reason):
+            transformed_response = _ChatChunkResponse(
+                id=None,
+                created=int(time.time()),
+                model=None,
+                choices=[
+                    _ChatChoiceDelta(
+                        index=0,
+                        delta=_ChatDeltaMessage(
+                            role="assistant",
+                            content=message_content,
+                        ),
+                        finish_reason=finish_reason,
+                    )
+                ],
+            )
+
+            if is_pydantic_v1:
+                return json.loads(transformed_response.json())
+            else:
+                return transformed_response.model_dump(mode="json")
+
+        def _convert(chunk):
+            if isinstance(chunk, str):
+                message_content = chunk
+                finish_reason = None
+            elif isinstance(chunk, AIMessageChunk):
```

Comment:
`AIMessageChunk` has `id`:

https://api.python.langchain.com/en/latest/messages/langchain_core.messages.ai.AIMessageChunk.html

Is it possible to pass it to `_ChatChunkResponse`?

Reply:
Yes. Updated PR. id is added.